usaspending_api/download/management/commands/download_sqs_worker.py

```diff
@@ -1,59 +1,160 @@
-                # waiting for the timeout window to expire
-                for message in messages:
-                    try:
-                        message.change_visibility(VisibilityTimeout=0)
-                    except botocore.exceptions.ClientError:
-                        # TODO: check existence instead of catching error
-                        continue
+                update_download_record(download_job_id, "ready")
+                release_sqs_message(CURRENT_MESSAGE)
+                time.sleep(MONITOR_SLEEP_TIME)  # Wait. System might be shutting down.
+
+
+def poll_queue(long_poll_time):
+    # Future TODO: allow different types of queues (redis, etc)
+    return poll_sqs_for_message(settings.BULK_DOWNLOAD_SQS_QUEUE_NAME, long_poll_time)
+
+
+def poll_sqs_for_message(queue_name, wait_time):
+    """ Returns 0 or 1 message from the queue"""
+    try:
+        queue = get_sqs_queue_resource(queue_name=queue_name)
+        sqs_message = queue.receive_messages(
+            WaitTimeSeconds=wait_time,
+            MessageAttributeNames=["All"],
+            VisibilityTimeout=DEFAULT_VISIBILITY_TIMEOUT,
+            MaxNumberOfMessages=1,
+        )
+    except botocore.exceptions.ClientError:
+        write_to_log(message="SQS connection issue. Investigate settings", is_error=True)
+        raise SystemExit(1)
+
+    return sqs_message[0] if sqs_message else None
+
+
+def create_and_start_new_process(download_job_id):
+    download_app = Process(
+        name="Download Service Worker Proccess", target=download_service_app, args=(download_job_id,)
+    )
+    download_app.start()
+
+    return download_app
+
+
+def download_service_app(download_job_id):
+    download_job = retrive_download_job_from_db(download_job_id)
+    write_to_log(message="Starting new Download Service App with pid {}".format(os.getpid()), download_job=download_job)
+
+    # Retrieve the data and write to the CSV(s)
+    try:
+        csv_generation.generate_csvs(download_job=download_job)
+    except Exception as e:
+        write_to_log(message="Caught exception", download_job=download_job, is_error=True)
+        return 11  # arbitrary positive integer
+
+    return 0
+
+
+def remove_message_from_sqs(message):
+    if message is None:
+        return
+    try:
+        message.delete()
+    except Exception as e:
+        write_to_log(message="Unable to delete SQS message. Message might have previously been released or removed")
+
+
+def set_sqs_message_visibility(message, new_visibility):
+    if message is None:
+        write_to_log(message="No SQS message to modify. Message might have previously been released or removed")
+        return
+    try:
+        message.change_visibility(VisibilityTimeout=new_visibility)
+    except botocore.exceptions.ClientError as e:
+        write_to_log(message="Unable to set VisibilityTimeout. Message might have previously been released or removed")
+
+
+def release_sqs_message(message):
```

Comment:
doc string
(i.e. clarify that this will make the message visible in the queue again, immediate, for any other consumer to consume)